### ISSUE TITLE
adding firstMap()

### ISSUE THREAD

**Issue by chrisguttandin** [cartant/rxjs-etc#89]
Hi Nicholas,

what do you think about adding a `firstMap()` operator? I often use a pattern like this in my code:

```ts
someObservable$
    .pipe(
        first(),
        mergeMap(() => {
            // ...
        })
    )
```

But sometimes I end up using `switchMap()` or `concatMap()` instead of `mergeMap()`. I think it doesn't matter which `xxxMap()` operator gets used after the `first()` operator and it always makes me wonder if it would be easier to reason about the code if there was an operator which just hides that.

```ts
someObservable$
    .pipe(
        firstMap(() => {
            // ...
        })
    )
```

If you think it's a useful addition, I would be happy to provide a PR for it.

**Comment by cartant** [comment]
Hmm. I'm not sure. If someone unfamiliar with `firstMap` were to encounter it in source, would they assume that if flattens rather than effects a higher-order observable - i.e. that it's not `...pipe(first(), map(value => /* map value to an ObservableInput */)`? What if the mapped value was an array (or an iterable)? Reading it, it wouldn't be obvious that the resultant observable would emit the arrays elements rather than the array itself.

**Comment by chrisguttandin** [comment]
Yes, you're right. I did not realize that the name would be misleading.

Do you have an idea for a better name. If not I'm also happy to just close this. I was just an idea.
